app/src/main/java/io/github/muntashirakon/setedit/EditorActivity.java

```diff
@@ -155,7 +155,7 @@ public void onCreate(Bundle bundle) {
                 if (isGranted) {
                     addNewItemDialog();
                 } else {
-                    EditorUtils.displayGrantPermissionMessage(this);
+                    EditorUtils.displayGrantPermissionMessage(this, ((SettingsRecyclerAdapter) adapter).getSettingsType());
```

Comment:
You may want to store `((SettingsRecyclerAdapter) adapter).getSettingsType()` to another variable since there are more than one access in this scope.